compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1832,7 +1832,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
             (msg, None)
         } else if ident.name == kw::SelfUpper {
-            ("`Self` is only available in impls, traits, and type definitions".to_string(), None)
+            if opt_ns.is_none() {
```

Comment:
This is kinda strange on first impression, but it literally says above that "// `None` indicates a module path in import"... so :shrug:

Reply:
couldn't hurt